recipe/build.sh

```diff
@@ -25,6 +25,9 @@ cmake -DCMAKE_INSTALL_PREFIX="${INSTALL_PREFIX}" \
       -DCMAKE_BUILD_TYPE=${BUILD_TYPE} \
       -DBUILD_TESTS=${BUILD_TESTS} \
       -DBUILD_BENCHMARKS=${BUILD_BENCHMARKS} \
+      -DCPM_LOCAL_PACKAGES_ONLY=ON \
+      -DCPM_Thrust_SOURCE=${PREFIX} \
+      -DCPM_spdlog_SOURCE=${PREFIX} \
```

Comment:
We really shouldn't need these. It looks like RMM doesn't have the conda environment handling snippet that other RAPIDS libraries do (https://github.com/rapidsai/cudf/blob/branch-22.08/cpp/CMakeLists.txt#L122) which adds the conda env to the cmake prefix path to have it searched in `find_package` calls automatically.

I also think that you're pointing to these packages as if they're source directories that should be added via an `add_subdirectory` type of call, but they should be installed packages found via `find_package`. I think we could use `Thrust_ROOT` and `spdlog_ROOT` instead here.

Reply:
because CPM first calls `find_package`

Reply:
```suggestion
      -DThrust_ROOT=${PREFIX} \
      -Dspdlog_ROOT=${PREFIX} \
```

Reply:
@kkraus14 @robertmaynard Unfortunately it doesn't work...

Reply:
Looks like spdlog worked but thrust didn't:
```
-- CPM: using local package spdlog@1.8.5
-- NO RAPIDS_VERSION for spdlog
CMake Error at build/cmake/CPM_0.35.0.cmake:584 (message):
  CPM: Thrust not found via find_package(Thrust 1.15.0)
Call Stack (most recent call first):
  build/cmake/CPM_0.35.0.cmake:275 (CPMAddPackage)
  build/_deps/rapids-cmake-src/rapids-cmake/cpm/find.cmake:152 (CPMFindPackage)
  build/_deps/rapids-cmake-src/rapids-cmake/cpm/thrust.cmake:72 (rapids_cpm_find)
  cmake/thirdparty/get_thrust.cmake:19 (rapids_cpm_thrust)
  cmake/thirdparty/get_thrust.cmake:26 (find_and_configure_thrust)
  CMakeLists.txt:60 (include)
  ```

Reply:
nvm, that's spdlog 1.8.5 being grabbed by cpm as opposed to spdlog 1.10.0 in the conda environment

I'm guessing this is related to cmake prefix paths?

Reply:
It's confusing, I thought conda would be able to find any cmake configs that come with the installed packages (thrust/spdlog in this case)?

Reply:
@robertmaynard any insights here?

Reply:
```
2022-05-26T17:22:44.7681580Z +CMAKE_PREFIX_PATH=$PREFIX:$BUILD_PREFIX/x86_64-conda-linux-gnu/sysroot/usr
```

This should be setting the search paths appropriately, but I think it's supposed to be semicolon separated as opposed to colon separated?

Reply:
Looks like this may need to be fixed? https://github.com/conda-forge/ctng-compiler-activation-feedstock/blob/7ac7228866a63b3cd0f9db7ab9b7e362bf58650d/recipe/activate-gcc.sh#L92-L99

Reply:
That is correct, You should use `:` for the the env variable `CMAKE_PREFIX_PATH`

>  I thought Conda's CMake would be able to find any cmake configs that come with the installed packages

It should. I will reproduce locally, and report back

Reply:
Interesting, so a bad `CMAKE_PREFIX_PATH` would make the search fail...

Anyway, I sent a fix: https://github.com/conda-forge/ctng-compiler-activation-feedstock/pull/76